tests/regression/test_mesh_generation.py

```diff
@@ -421,3 +421,11 @@ def test_changing_default_reorder_works(reorder):
         assert m._did_reordering == reorder
     finally:
         parameters["reorder_meshes"] = old_reorder
+
+
+@pytest.mark.parametrize("kind",
+                         ["default", "crossed"])
+def test_boxmesh_diagonal(kind):
+    m = BoxMesh(1, 1, 1, 1, 1, 1, diagonal=kind)
+    m.init()
+    assert m.num_cells() == 6 or 5
```

Comment:
This logic doesn't do what you want. It evaluates as `(m.num_cells() == 6) or 5` and then `5` is interpreted in a boolean context, so this is always true.

You really want to assert that you got the correct number of cells too, so try:

```python
@pytest.mark.parametrize("kind,num_cells", [("default", 6), ("crossed", 5)])
def test_boxmesh_kind(kind, num_cells):
    ...
    assert m.num_cells() == num_cells
```

Reply:
Oh, no. You are right! I just pushed the corrected testing in the latest commit. Thanks, Lawrence!